src/klein/app.py

```diff
@@ -329,12 +330,33 @@ def run(self, host, port, logFile=None):
 
         @param logFile: The file object to log to, by default C{sys.stdout}
         @type logFile: file object
+
+        @param proto: protocol string for endpoint, can be either "tcp" or "ssl"
+        @type proto: str
+
+        @param privateKey: filepath to private key for ssl
+        @type privateKey: str
+
+        @param certificate: filepath to certificate for ssl
+        @type certificate: str
+
+        @param dhParameters: filepath to dh parameters for ssl
+        @type dhParameters: str
         """
         if logFile is None:
             logFile = sys.stdout
 
         log.startLogging(logFile)
-        reactor.listenTCP(port, Site(self.resource()), interface=host)
+        server_string = "{}:{}:interface={}".format(proto, port, host)
+
+        if privateKey and certificate:
+            server_string += ":privateKey={}:certKey={}".format(privateKey,
+                                                                certificate)
+        if dhParameters:
+            server_string += ":dhParameters={}".format(dhParameters)
+
+        endpoint = endpoints.serverFromString(reactor, server_string)
+        endpoint.listen(Site(self.resource()))
```

Comment:
> However, I otherwise agree with your other option, which is to create a new method. That's also fine by me.

On second thought I realized that one drawback of new method is that it will create two ways of doing same thing, so this may add some confusion. Aside from that some server specifications will work but probably don't apply to Klein, e.g. "unix:address=/var/run/web.sock:lockfile=1" will be valid string that could perhaps possibly work for serverFromString but probably doesn't make much sense for Klein. 

Using endpoints directly may allow to limit arguments to those relevant for TCP and SSL and keep only one way of doing things.